Feat/GitHub commit dataset

abhisomala opened the pull request:
creates csv and makes atlas map using github repo url

Commit:
github commit messages to atlas map

Commit:
makes map for git repo comments

abhisomala committed:
makes map for git repo comments

abhisomala committed:
Update github_commits.py

abhisomala committed:
Update github_commits.py

AndriyMulyar commented on maps/github_commits.py:
```diff
@@ -0,0 +1,96 @@
+
+
+if __name__ == "__main__":
+    repo_url = input("URL of Github Repository: ")
```
You should use python tmpdirectory for storing the cloned down git repository.
https://docs.python.org/3/library/tempfile.html

AndriyMulyar replied:
not a place on the filesystem. the tmpfile package will automatically pick the right temporary storage place on each OS.

Use a `context manager` to ensure the temp directory get's cleaned up afterwords.

AndriyMulyar commented on maps/github_commits.py:
```diff
@@ -0,0 +1,96 @@
+
+
+    dataset = AtlasDataset(
+        "github-commit-dataset",
```
You should try to infer this title from the github repo name

AndriyMulyar replied:
also, be explicit with identifier='datasetname'

abhisomala committed:
Update github_commits.py

abhisomala committed:
Update github_commits.py

abhisomala committed:
incorporated tmp directory

Adaptive map and csv file name, creates a tmp directory that (should) deletes itself after its done

abhisomala committed:
Update github_commits.py

Works with the large github repos such as the linux kernal with over 1 million commits

rguo123 commented on maps/github_commits.py:
```diff
@@ -0,0 +1,116 @@
+if __name__ == "__main__":
+    repo_urls = input("Enter GitHub Repository URLs separated by commas: ").split(',')
+    all_commits = [] #An attempt at allowing multiple github repos, however it doesnt work
+    repo_names = [] #still trying to figure it out - saying that there isnt enough data points (even though there is)
```
Nit: remove wip comments from script?
In general, may be worth cleaning up some of the code comments so formatting is consistent

rguo123 commented on maps/github_commits.py:
```diff
@@ -0,0 +1,116 @@
+    dataset.add_data(data=data)
+
+    map = dataset.create_index(
+        indexed_field='message',  #Indexs by message
```
Nit typo 'indexes by message'

rguo123 commented on maps/github_commits.py:
```diff
@@ -0,0 +1,116 @@
+    return commit_list
+
+# Saves commits to a CSV file
+def save_commits_to_csv(commits, csv_filename):
```
It would be nice if this was an optional step in pipeline. I think the default should be to go directly from reading in the commit history into atlas rather than having first to save it as a csv

abhisomala committed:
Optional step to create CSV and nit updates
